### PR TITLE
fix(parser): follow latest estree spec on ExportAllDeclaration

### LINKED ISSUE #97: Named export
Hi,

It seems that Meriyah does not comply with the AST standard when having named export, e.g. **export * as x from 'a'**.

First of all, it creates a _ExportNamedDeclaration_ where it stored an _ExportNamespaceSpecifier_ in the _specifiers_ array. The last one is unknown according to the specifications. If you compare it to Acorn, which seems to do it the right way, then you should create an _ExportAllDeclaration_ and then set the _exported_ property to 'x'.

If you look at the specifications, then you will see that this is the correct way to do it:

https://github.com/estree/estree/blob/master/es2020.md#exportalldeclaration

-Thomas

### PR DESCRIPTION
**Pull request by 3cp**
closes #97, #99

-----
Prettier touches few lines of code.

### CODE REVIEW

File: src/estree.ts [hunk 1]
```diff
@@ -415,12 +413,9 @@ export interface EmptyStatement extends _Node {
 export interface ExportAllDeclaration extends _Node {
   type: 'ExportAllDeclaration';
   source: Literal;
+  exported: Identifier | null;
 }
 
-export interface ExportNamespaceSpecifier extends _Node {
-  type: 'ExportNamespaceSpecifier';
```
Comment (KFlash) [inline src/estree.ts:421]: Has this AST node been removed from the ESTree specs?

Comment (3cp) [reply src/estree.ts:421]: Yes. It is not in estree spec master anymore.

Comment (KFlash) [reply src/estree.ts:421]: I see. What about [AnonymousDefaultExportedFunctionDeclaration](https://github.com/estree/estree/blob/f9d7bd7e6b8d9476f2eaa71d6ca279dff8d49d82/spec.estree#L799) ?

**Review by KFlash — APPROVED** [review]

**Comment by KFlash** [comment]
@3cp @aladdin-add can you do a new NPM release?

File: src/estree.ts (hunk 1 above)
Comment (3cp) [reply src/estree.ts:421]: We seem handled those with normal FunctionDeclaration and ClassDeclaration.

**Comment by 3cp** [comment]
I can do the release. Do you mind I add scripts for npm version to generate changelog (we don't have yet), and then publish?

Like this:

```
    "preversion": "npm test",
    "version": "standard-changelog && git add CHANGELOG.md",
    "postversion": "git push && git push --tags && npm publish",
```

**Comment by KFlash** [comment]
No problem if you want to add that 👍

**Pull request merged**